pkg/ottl/parser.go

```diff
@@ -27,6 +27,27 @@ type Parser[K any] struct {
 	telemetrySettings component.TelemetrySettings
 }
 
+// Statement is an interface that represents a parsed OTTL Statement
+type Statement[K any] interface {
+	// Execute is a function that will execute the statement's function if the statement's condition is met.
+	// Returns true if the function was run, returns false otherwise.
+	// If the statement contains no condition, the function will run and true will be returned.
+	Execute(ctx K) bool
```

Comment:
Could we have this function return the output of the execution? `ExprFunc` has a return type of `interface{}`, so user-written functions could return a value. I could see this potentially being used in cases like the Filter Processor or Span Metrics Processor where telemetry data is added or removed and we want to write statements to perform that work, but need to communicate the result with the calling component.

Reply:
Yes great idea. I'll update the function to return both.

Reply:
Done

Reply:
@kentquirk agreed that there isn't any defined use case yet, but when I tried to implement `drop` returning a value from the function was important.  Hoping to get back to that function once some of the other refactors are handled.